### ISSUE TITLE
Scene support

### ISSUE THREAD

**Issue by jczwo** [NorthernMan54/homebridge-yamaha-home#6]
Your plugin is working very well. Thank you! The only thing I miss is scene support. As you integrated switches for changing inputs, maybe it's possible to do the same for scenes. Have a look at: https://github.com/codetheweb/homebridge-yamaha-scene

**Comment by NorthernMan54** [comment]
@jczwo I wonder if @TommyCardello would be interested in adding scenes as well?   He did the work on changing inputs.

**Comment by TommyCardello** [comment]
I think I've just done it in 5 minutes (using the 2nd method described above) 😁
I can't believe it was that easy, so I'll test it more, to be sure that it works as expected.

**Comment by jczwo** [comment]
Hi @TommyCardello,

On my receiver a have two scenes, both using HDMI 1. They only differ in sound settings. As far as I understand, this would not work with your current solution right?

I would be great if we can find some getScene() code and integrate it into yamaha-nodejs.

**Comment by TommyCardello** [comment]
@jczwo I'm 90% sure it will work. Make sure that you use a unique name for different scenes, but the same input it's okay.
I won't create a pull request until I get @NorthernMan54 opinion on a switch status update.

However, my solution should work for you right now.
You can try it by using my work: https://github.com/TommyCardello/homebridge-yamaha-home

Check out readme in my fork. Switch will work anyways, but you need to specify input in the config file, and switch will update its status correctly.

Anyway, I'll create a pull request to this repository later, when I'm sure it's ready. So, it's up to you. You can wait until this feature will be released here or you can try it in my fork.

To check is it possible to get scene info, I need to dive into yamaha-nodejs development. That will take time because I didn't work with that plugin before.
For now, if you specify input in a config, everything works perfectly.

**Comment by TommyCardello** [comment]
@NorthernMan54  I agree. To me, it looks pretty simple. All user have to do is to add "name" property for a scene and "setInputTo" property. It will work as is. But I'll try to think about a better way. Without mandatory "setInputTo" property if used scenes instead of switches. I'll do more research because if I pull it now and later I will change how the code works, it will be confusing. 

However, I'm looking into adding functionality to obtain scene name from receiver to yamaha-nodejs.
I don't know where should I start.

Option 1:
I found TCP command in YNCA protocol documentation. `@MAIN:SCENENAME=?`
I'm not familiar with it, I need to find out how to make it as xml request, so it can be added to yamaha-nodejs.

Option 2:
Scenes functionality is available in iOS app. I'm not sure, how to find how it works. But I think this way is the best.

**Comment by NorthernMan54** [comment]
@TommyCardello The easiest way to find out how it works, is to look at the web site built into the receiver with a browser, then use the browser debugging tools to determine what the messages are that control scenes.  It takes a bit of effort, but should give you what is needed.

**Comment by TommyCardello** [comment]
Thanks for the hint.
I've checked everything, the xml "PUT" request with scene number exists in chrome dev tools. But receiver does not return scene number/name back. In XML "GET" request which follows after "PUT" request, there is no information about a scene.
That's how it works in web interface.

In iOS app scene name does not refresh after I change it on receiver directly.

To me, it looks like it's not possible to get scene number back from the receiver.

But that TCP command in YNCA protocol `@MAIN:SCENENAME=?`... I was sure that's the command I'm looking for.

**Comment by NorthernMan54** [comment]
@TommyCardello Sir, I have to agree with you.  I looked at my yamaha this morning and am seeing that with the current API, we don't have access to the scene name.

**Comment by TommyCardello** [comment]
@NorthernMan54 Thank you for confirming it. 

I'm glad I was able to contribute to this project. 
I will clean up the code which I implemented and will submit a pull request later.

I don't feel like it's too complicated for a user to specify input name for a scene manually.
The code works even without that, but with specifying input, switch status updates correctly. 

Anyway, I used  3 different plugins and now all I need can be done with just one. 

P.S.  I made some research on YNCA protocol and YNCA command ` @MAIN:SCENENAME=?`. For some receivers, this command can be send by TCP and receiver should return scene name. I've seen some python plugins for that. But I'm not familiar with TCP and python, so currently it's beyond my capabilities and I'm not sure that this command will work for all receivers.

**Comment by jczwo** [comment]
@TommyCardello It seems that my Receiver needs a <Scene_Sel></Scene_Sel> Tag instead of <Scene_Load></Scene_Load>. Could you verify if <Scene_Sel> works for you too? I would like to see your work integrated into this repository. :-)

**Comment by NorthernMan54** [comment]
Just published this, please give it a try
